controllers/provider-aws/pkg/cmd/options.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	controlplanecontroller "github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/controller/controlplane"
+	infrastructurecontroller "github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/controller/infrastructure"
+	controlplanewebhook "github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/webhook/controlplane"
+	"github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/webhook/controlplaneexposure"
+	"github.com/gardener/gardener-extensions/pkg/controller/cmd"
+	extensionscontrolplanecontroller "github.com/gardener/gardener-extensions/pkg/controller/controlplane"
+	extensionsinfrastructurecontroller "github.com/gardener/gardener-extensions/pkg/controller/infrastructure"
+	extensioncontrolplanewebhook "github.com/gardener/gardener-extensions/pkg/webhook/controlplane"
+)
+
+// ControllerSwitchOptions are the cmd.ControllerSwitchOptions for the provider controllers.
+func ControllerSwitchOptions() *cmd.ControllerSwitchOptions {
+	opts := cmd.NewControllerSwitchOptions(
+		cmd.ControllerSwitch(extensionsinfrastructurecontroller.ControllerName, infrastructurecontroller.AddToManager),
```

Comment:
Didn't we agree that we want to hold a map of controller names to their `AddToManager` functions in `controllers/provider-*/pkg/controller/controller.go`, and simply iterate over it here (similar for webhooks)?

This would prevent the double maintenance of the list of controllers and make it easier when adding new controllers to a provider.

Right now, the [list in `../controllers.go`](https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-aws/pkg/controller/controller.go) still exists, and another list has been introduced here.

Reply:
Since we'll only use the `cmd.ControllerSwitchOptions` actually having `AddToManager` in the controllers is unused. I removed that and now we again only have one list to maintain.

Reply:
OK, thank you!